sdk/src/sharedTest/java/io/customer/sdk/api/HttpRequestRunnerTest.kt

```diff
@@ -192,6 +192,18 @@ class HttpRequestRunnerTest : BaseTest() {
         actual shouldBeEqualTo expected
     }
 
+    @Test
+    fun performHttpRequest_given400_expectNotPausedHttpRequestsAndReturnFailure(): Unit = runBlocking {
```

Comment:
Good test to add. 

Does this test exist for the iOS SDK PR? I think it's a valuable test.